prov/util/src/util_mr_map.c

```diff
@@ -82,6 +83,8 @@ int ofi_mr_map_insert(struct ofi_mr_map *map, const struct fi_mr_attr *attr,
 	if (map->mode & FI_MR_PROV_KEY)
 		item->requested_key = map->key++;
 
+	key_is_optional = !(attr->access & (FI_REMOTE_READ | FI_REMOTE_WRITE));
```

Comment:
Should `FI_READ` and `FI_WRITE` also be included?

Reply:
FI_READ and FI_WRITE are for local access only (the result buffer for fi_read, and the source buffer for fi_write).